### DISCUSSION TITLE
RTSP url in html webpage

### DISCUSSION THREAD

**Discussion by Pedro-Leitek** [AxisCommunications/discussions#98] [VAPIX] [answered]
Hello

I would like to know the tutorial to implement a my axis stream in a html page. It is known that I can't put rtsp url directly on "src" attribute of video or img tag. I tried using the api with "/axis-cgi/mjpg/video.cgi" extension, but I realized that the quality is very bad. I would like to have a stream equal to the one in axis camera main page (Live player). Any idea on how to implement this?

Thanks

**Comment by mattias-kindborg-at-work** [comment, marked as answer]
Hi @Pedro-Leitek. We've got some excellent open source repositories covering this, please see [Media Stream Player JS](https://github.com/AxisCommunications/media-stream-player-js) and [Media Stream Library JS](https://github.com/AxisCommunications/media-stream-library-js).

**Comment by Pedro-Leitek** [comment]
@mattias-kindborg-at-work I was able to use de media Stream Library JS. I can see the stream but everytime the window/tab of the browser is not in focus because I've switched to another tab or minimized the browser, I see a delay on the stream. It is like if the stream freezed when the tabs were switched and resumed when the tab was opened again. I would like to know what can I do to see the stream live (real-time) no matter how long the tab is not focused. Thanks

**Reply by mattias-kindborg-at-work** [reply]
I think you should start a discussion or create an issue in Media Stream Library JS, as they will have a better chance at helping you than I have.
